src/cascadia/TerminalApp/TabHeaderControl.idl

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace TerminalApp
+{
+    delegate void HeaderTitleWantsToChangeArgs(String title);
+
+    [default_interface] runtimeclass TabHeaderControl : Windows.UI.Xaml.Controls.UserControl, Windows.UI.Xaml.Data.INotifyPropertyChanged
+    {
+        String Title { get; set; };
+
+        TabHeaderControl();
+        void SetZoomIconVisibility(Windows.UI.Xaml.Visibility state);
+        void ConstructTabRenameBox();
```

Comment:
nit: Should we name this something like `BeginRename`? Caller doesn't need to know it's constructing anything